inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -107,30 +107,46 @@ def article_exists(obj, eng):
 
 
 @with_debug_logging
-def is_being_harvested_on_legacy(record):
-    """Return True if the record is being harvested on Legacy.
+def belongs_to_relevant_category(obj, eng):
+    """Return True if the record is Core or Non-Core.
+
+    Arguments:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        bool: ``True`` if the ingested article belongs to a relevant category.
 
-    If the record belongs to one of the CORE arXiv categories then it
-    is already being harvested on Legacy.
     """
-    arxiv_categories = get_arxiv_categories(record)
-    legacy_categories = current_app.config.get(
-        'ARXIV_CATEGORIES_ALREADY_HARVESTED_ON_LEGACY', [])
+    return get_coreness(obj.data) is not None
 
-    return len(set(arxiv_categories) & set(legacy_categories)) > 0
 
+def get_coreness(record):
+    """Return the coreness of the record harvested.
 
-@with_debug_logging
-def already_harvested(obj, eng):
-    """Check if record is already harvested."""
-    if is_being_harvested_on_legacy(obj.data):
-        obj.log.info((
-            'Record with arXiv id {arxiv_id} is'
-            ' already being harvested on Legacy.'
-        ).format(arxiv_id=get_arxiv_id(obj.data)))
+    Args:
+        record(dict): the record harvested.
+
+    Return:
+        True if the record belongs to one of the CORE arXiv categories.
+        False, if it belongs to one of the Non-CORE categories.
+        None, in the other cases.
+
+    """
+    arxiv_categories = set(get_arxiv_categories(record))
+    relevant_categories = current_app.config.get('ARXIV_CATEGORIES', [])
+
+    if arxiv_categories & set(relevant_categories['core']):
         return True
+    elif arxiv_categories & set(relevant_categories['non-core']):
+        return False
+    else:
+        return None
```

Comment:
I don't like this `True`-`False`-`None` triplet :/, maybe we can use an enum instead, something like:
```python
from enum import Enum
class Coreness(Enum):
    core = True
    non_core = False
    non_relevant = None

# then return one of the properties in get_coreness
def get_coreness(...):
    ...
    if ...:
        coreness = Coreness.core
    elif ...:
        coreness = Coreness.non_core
    else:
        coreness = Coreness.non_relevant
    return coreness
```

It requires `enum34` on python 2 though, that we indirectly require, but I find it way more readable.

Then when setting it you just need:
```python
coreness = get_coreness(obj.data)
obj.extra_data['core'] = coreness.value
```

Reply:
Great, that was the feedback I wanted! I don't like it too.